src/clj/puppetlabs/services/ca/certificate_authority_service.clj

```diff
@@ -35,7 +42,19 @@
          auth-handler
          puppet-version)
        {:normalize-request-uri true})
-     (assoc context :auth-handler auth-handler)))
+     (watch-protocol/add-watch-dir! watcher
+                                    (fs/parent ca-crl-file)
+                                    {:recursive true})
```

Comment:
For reference, I submitted a ticket for the work to add support for non-recursive directory watching to the tk-filesystem-watcher service, [TK-448](https://tickets.puppetlabs.com/browse/TK-448).  It might be reasonable to consider merging this ahead of that work getting done.